custom_components/econet300/number.py

```diff
@@ -115,11 +115,23 @@ def can_add(key: str, coordinator: EconetDataCoordinator):
     return coordinator.has_data(key) and coordinator.data[key]
 
 
-def apply_limits(desc: EconetNumberEntityDescription, limits: Limits):
-    """Set the native minimum and maximum values for the given entity description."""
-    desc.native_min_value = limits.min
-    desc.native_max_value = limits.max
-    _LOGGER.debug("Apply limits: %s", desc)
+def apply_limits(
+    desc: EconetNumberEntityDescription, limits: Limits
+) -> EconetNumberEntityDescription:
+    """Create a new EconetNumberEntityDescription with updated limits."""
+    updated_desc = EconetNumberEntityDescription(
```

Comment:
Was this function tested? Because its seams instead of updating OBJECT it creates new OBJECT and returns it